### PR TITLE
fix(permissions): to match inferred data fence (view when manage)

### PR DESCRIPTION
**Pull request by tdeekens**
#### Summary

This pull request fixes data fencing matching the inferred permission: view should be inferred by manage as with general permissions.

#### Description

I went commit by commit and review should also be possible as such.

### CODE REVIEW

File: packages/permissions/src/utils/has-permissions.ts [hunk 1]
```diff
@@ -313,10 +319,17 @@ export const hasSomeDataFence = (options: TOptionsForAppliedDataFence) => {
...
-      ).find(
-        ([dataFenceName, dataFenceValue]) =>
-          dataFenceValue && dataFenceName === toCanCase(demandedDataFence.name)
-      );
+      ).find(([dataFenceName, dataFenceValue]) => {
+        // Either the manage permission matches or the
+        // a demanded view permission is implied by a manage permission.
+        const doesDataFenceMatchByName =
+          dataFenceName === toCanCase(demandedDataFence.name) ||
+          dataFenceName === toCanCase(toManageCase(demandedDataFence.name));
```
Comment (tdeekens) [inline packages/permissions/src/utils/has-permissions.ts]: The fix. The above is just moving logic around to be reusable.

File: packages/permissions/src/utils/has-permissions.spec.tsx [hunk 2]
```diff
@@ -290,8 +318,60 @@ describe('hasSomeDataFence', () => {
...
+          demandedDataFences: [
+            {
+              type: 'store',
+              group: 'orders',
+              name: 'ViewOrders',
+            },
+          ],
+          selectDataFenceData: () => ['store-1'],
+        })
+      ).toBe(true);
```
Comment (adnasa) [inline packages/permissions/src/utils/has-permissions.spec.tsx]: yes! 💯

File: packages/permissions/src/utils/has-permissions.ts (hunk 1 above)
Comment (adnasa) [reply packages/permissions/src/utils/has-permissions.ts]: to clarify:

```
demandedDF: VIEW
dataFenceName VIEW
output: true

demandedDF: MANAGE
dataFenceName VIEW
output: false

demandedDF: VIEW
dataFenceName MANAGE
output: true

demandedDF: MANAGE
dataFenceName MANAGE
output: true
```

Comment (tdeekens) [reply packages/permissions/src/utils/has-permissions.ts]: Yes.

**Comment by tdeekens** [comment]
I don't reproduce this test failure locally. Even with `CI=true` not. Hm.

File: packages/permissions/src/utils/has-permissions.spec.tsx [hunk 3]
```diff
@@ -33,7 +33,7 @@ describe('hasPermission', () => {
       demandedPermission = 'ManageProducts';
       actualPermissions = { canManageProducts: true };
     });
-    it('should return true', () => {
+    it('should indicate the permission the permission matches', () => {
```
Comment (emmenko) [inline packages/permissions/src/utils/has-permissions.spec.tsx]: Duplicated words?

File: packages/permissions/src/utils/has-permissions.ts [hunk 4]
```diff
@@ -66,7 +66,11 @@ type TOptionsForAppliedDataFence = {
 
 // Build the permission key from the definition to match it to the format coming
 // from the API.
```
Comment (emmenko) [inline packages/permissions/src/utils/has-permissions.ts]: This comment refers to the `toCanCase` function

File: packages/application-shell/src/components/application-shell/application-shell.spec.js [hunk 5]
```diff
@@ -1142,7 +1142,7 @@ describe('when navbar menu items do not match given data fences', () => {
               value: 'usa',
               type: 'store',
               group: 'orders',
-              name: 'canManageOrders',
+              name: 'canViewOrders',
```
Comment (emmenko) [inline packages/application-shell/src/components/application-shell/application-shell.spec.js]: Thanks! Was this related to the changes in this PR or was it wrong before?

Comment (tdeekens) [reply packages/application-shell/src/components/application-shell/application-shell.spec.js]: It related to the changes and was kind of right but also wrong. data fence permissions should infer view from manage. Just like with any other permission.

**Pull request merged**